### PR TITLE
Key generation and loading examples using disk to store the key

### PR DESCRIPTION
**Pull request by dgarske**
Leverage wolfSSL filesystem abstraction.

TODO: Refactor the other filesystem interactions to leverage wolf XFOPEN type API's.

ZD 11163

### CODE REVIEW

File: examples/keygen/keyload.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        fileSz = XFTELL(f);
+        XREWIND(f);
+        if (fileSz > sizeof(newKey.priv) + sizeof(newKey.pub)) {
+            rc = BUFFER_E;
+            goto exit;
+        }
+        printf("Reading %d bytes from %s\n", (int)fileSz, inputFile);
+        
+        XFREAD(&newKey.pub, 1, sizeof(newKey.pub), f);
+        XFREAD(&newKey.priv, 1, sizeof(newKey.priv), f);
```
Comment (dgarske) [inline examples/keygen/keyload.c:100]: This could / should be read remainder... Also I can do size checks....

File: examples/keygen/keygen.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    }
+    printf("Created new key (pub %d, priv %d bytes)\n",
+        newKey.pub.size, newKey.priv.size);
+
+    /* Save key as encrypted blob to the disk */
+#if !defined(WOLFTPM2_NO_WOLFCRYPT) && !defined(NO_FILESYSTEM)
+    f = XFOPEN(outputFile, "wb");
+    if (f != XBADFILE) {
+        XFWRITE(&newKey.pub, 1, sizeof(newKey.pub), f);
+        XFWRITE(&newKey.priv, 1, sizeof(newKey.priv), f);
```
Comment (dgarske) [inline examples/keygen/keygen.c:113]: This doesn't have to write the entire priv buffer, just `newKey.priv.size + sizeof(UINT16)`.

File: examples/keygen/keyload.c [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+        /* sanity check the sizes */
+        if (newKey.pub.size != sizeof(newKey.pub) || newKey.priv.size > sizeof(newKey.priv.buffer)) {
+            printf("Struct size check failed (pub %d, priv %d)\n", newKey.pub.size, newKey.priv.size);
+            rc = BUFFER_E; goto exit;
+        }
+    }
+    else {
+        printf("File %s not found!\n", inputFile);
+        goto exit;
```
Comment (elms) [inline examples/keygen/keyload.c:117]: nit: if file doesn't exist, rc is 0 and the process exits with 0. I'd probably also exit with non-0 on loading blob not supported also.

**Review by elms — APPROVED** [review]
One minor nit on `keyload` exit code. Great addition.

**Comment by dgarske** [comment]
@elms , fixed. Looking forward to your merge of these changes and possible cleanup to use your test key functions for reading/writing key blobs.

**Review by elms — APPROVED** [review]

**Pull request merged**